src/main/java/dev/galacticraft/mod/content/block/special/WebStringBlock.java

```diff
@@ -0,0 +1,229 @@
+        // Find a pos directly below the existing block (this) but above world bottom where the new block can be placed.
+        for (BlockPos checkPos = new BlockPos.MutableBlockPos(pos.getX(), pos.getY(), pos.getZ()); checkPos.getY() > level.getMinBuildHeight(); checkPos = checkPos.offset(0, -1, 0)) {
+            ItemInteractionResult result = checkCanBlockBePlaced(level, checkPos, player, stack, newState);
+            if (result != null) return result;
+        }
+        return null;
+    }
+
+    /// This method is called when the player tries to use something on a Web String block.
+    /// The default behaviour is extended by checking if that item is a glowstone torch, which
+    /// automatically hands the torch in the web. Web strings below the torch are broken.
+    @Override
+    protected ItemInteractionResult useItemOn(ItemStack stack, BlockState state, Level level, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+        ItemInteractionResult result = null;
+        // If using a glowstone torch on a web string, then it gets hung in the web string becoming a web torch block.
+        if (stack.is(GCItems.GLOWSTONE_TORCH)) {
+            result = convertToWebTorch(stack, state, level, pos, player);
+        // If using a Web string on the web string, then extend the whole column of web strings down if we can.
+        } else if (stack.is(GCBlocks.WEB_STRING.asItem())) {
+            result = extendBlockDown(stack, level, pos, player, defaultBlockState().setValue(WEB_STRING_PART, WebStringPart.BOTTOM));
+        // If using a web torch on the web string, then extend the whole column of web strings down if we can.
+        } else if (stack.is(GCBlocks.WEB_TORCH.asItem())) {
+            result = extendBlockDown(stack, level, pos, player, GCBlocks.WEB_TORCH.defaultBlockState());
+        }
+
+        if (result == null) {
+            result = super.useItemOn(stack, state, level, pos, player, hand, hit);
+        }
+        return result;
+    }
+
+    /// Manages the appearance of the web strings via the web string part property. Effectively top means it's attached
+    /// to something, bottom means it is the end dangling down, and middle means it is neither the top nor the bottom.
+    /// Top_Bottom means it is both the top and the only bit. The method also handles breaking of the web string
+    /// if the block above is broken.
+    @Override
+    protected BlockState updateShape(BlockState state, Direction direction, BlockState neighborState, LevelAccessor level, BlockPos pos, BlockPos neighborPos) {
+        // If above changed and it isn't a web string block (and in theory not a solid block) then it breaks.
+        if (direction == Direction.UP && !neighborState.is(GCBlocks.WEB_STRING)) {
+            return Blocks.AIR.defaultBlockState();
+            // If below changed
+        } else if (direction == Direction.DOWN) {
```

Comment:
Which line does this comment relate to?

If it is the line below then I suggest the following:

```suggestion
        } else if (direction == Direction.DOWN) { // If below changed
```